Src/Support/Google.Apis.Auth/OAuth2/JsonCredentialParameters.cs

```diff
@@ -34,63 +37,166 @@ public class JsonCredentialParameters
         /// <a href="https://console.developers.google.com">Google Developers Console</a>
         /// </summary>
-        [Newtonsoft.Json.JsonProperty("client_email")]
+        [JsonProperty("client_email")]
         public string ClientEmail { get; set; }
 
         /// <summary>
         /// Private Key associated with ServiceAccountCredential obtained from
         /// <a href="https://console.developers.google.com">Google Developers Console</a>.
         /// </summary>
-        [Newtonsoft.Json.JsonProperty("private_key")]
+        [JsonProperty("private_key")]
         public string PrivateKey { get; set; }
 
         /// <summary>
         /// Private Key ID associated with ServiceAccountCredential obtained from
         /// <a href="https://console.developers.google.com">Google Developers Console</a>.
         /// </summary>
-        [Newtonsoft.Json.JsonProperty("private_key_id")]
+        [JsonProperty("private_key_id")]
         public string PrivateKeyId { get; set; }
 
         /// <summary>
         /// Refresh Token associated with UserCredential created by
         /// <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">GCloud Auth Login</a>.
         /// </summary>
-        [Newtonsoft.Json.JsonProperty("refresh_token")]
+        [JsonProperty("refresh_token")]
         public string RefreshToken { get; set; }
+
+        /// <summary>
+        /// The STS audience associated with an ExternalAccountCredential.
+        /// </summary>
+        [JsonProperty("audience")]
+        public string Audience { get; set; }
+
+        /// <summary>
+        /// The STS subject token type associated with an ExternalAccountCredential.
+        /// </summary>
+        [JsonProperty("subject_token_type")]
+        public string SubjectTokenType { get; set; }
+
+        /// <summary>
+        /// The STS token exchange endpoint associated with an ExternalAccountCredential.
+        /// </summary>
+        [JsonProperty("token_url")]
+        public string TokenUrl { get; set; }
+
+        /// <summary>
+        /// The URL for the service account impersonation request associated with an ExternalAccountCredential.
+        /// </summary>
+        [JsonProperty("service_account_impersonation_url")]
+        public string ServiceAccountImpersonationUrl { get; set; }
+
+        /// <summary>
+        /// The endpoint used to retrieve the account related information 
+        /// associated with an ExternalAccountCredential.
+        /// </summary>
+        [JsonProperty("token_info_url")]
+        public string TokenInfoUrl { get; set; }
+
+        /// <summary>
+        /// The credential source associated with and ExternalAccountCredential.
+        /// </summary>
+        [JsonProperty("credential_source")]
+        public ExternalAccountCredentialSource CredentialSource { get; set; }
+
+        /// <summary>
+        /// Holder for the credential source parameters associated to external account credentials.
```

Comment:
Are we happy to assume we'll never need two different sets of parameters which use the same field but with different types?